README.md

```diff
@@ -257,7 +257,7 @@ Here are some examples:
   trainer = Trainer(logger=loggers.MLFlowLogger())
 
   # neptune
-  trainer = Trainer(logger=loggers.NeptuneLogger())
+  trainer = Trainer(logger=loggers.NeptuneLegacyLogger())
```

Comment:
Is the new `NeptuneLogger` backwards compatible with the old one?

If not, we should keep the current name and instead have the new one use a different name. The current one would be deprecated and eventually removed.

Reply:
Hi @carmocca @awaelchli @Borda! 

Marcin, Product Manager from Neptune on this side. 

The context behind this change is that we introduced a new API, however, the legacy one will continue to live for some time (we plan to support it for at least a year). We are in the process of slowly sunsetting it, but it's not "deprecated" yet - we are actively maintaining it. The new API is the new default one (documentation, onboarding, etc.).

We thought about two options:
• **Adding Logger using new API as a new Logger.**
This solution has a problem that the new Logger would have a misleading name (New/Next/2/Next) for the user and the user does not know which one should they use.
• **Renaming current one as Legacy and adding new Logger under the "default" name.**
This solution on the other hand has a problem of backward compatibility as you mentioned. With the initial design for the logger we hoped to minimize this problem. However, the points about catching outdated parameters through kwargs and some methods no longer available is not a pleasant thing for a user.

As you pointed out we could improve this. I would propose the following modification:
1) If the user has not updated neptune-client we inform them about this and point to NeptuneLegacyLogger. [This is partially done, but can be improved with messaging]
2) If the user's underlying project is not migrated to the new structure we inform them about this and point to NeptuneLegacyLogger. This is temporary only, new projects are using the new structure, we are in the process of migrating existing projects. [This is partially done, but can be improved with messaging]
3) Capture old parameters through kwargs to detect "Legacy usage" and act accordingly.
4) Add methods that were exposed by base NeptuneLogger (most of the interaction was actually through `logger.experiment.method`) to keep the API of the logger the same. If the user is using those methods we act in a similar way as in 3.

Let me know what you think about that and we will act accordingly :) 

I want the transition to be as painless for the users as possible, but on the other side we did change this API based on feedback as it was often very limiting to the users and we don't want to mislead new users into using the legacy API.

Reply:
> As you pointed out we could improve this. I would propose the following modification:

Sounds good to me 😃 
@carmocca @edenlightning what do you think?

Reply:
Ok, adapted to your  suggestions

Reply:
You could do something like this so users don't have to know about the different logger classes:

```python
class BaseNeptuneLogger: ...
class NeptuneLegacyLogger(BaseNeptuneLogger): ...
class NeptuneV2Logger(BaseNeptuneLogger): ...

def choose_logger(*args, **kwags) -> BaseNeptuneLogger:
    if on_latest_neptune_version() and latest_version_args(*args, **kwargs):
        return NeptuneV2Logger(*args, **kwargs)
    show_should_update_warning()
    return NeptuneLegacyLogger(*args, **kwargs)

NeptuneLogger = chooose_logger
```

The disadvantage of this approach is that docs will be shown under the `Neptune{Legacy,V2}Logger` classes.

Reply:
We've already decided that we'll take care of arguments passed to new logger.
If user will pass any deprecated arguments we'll inform him that his code is not adapted to the new one.